### DISCUSSION TITLE
extract text and shapes of vector pdf for a rectangular region

### DISCUSSION THREAD

**Discussion by sanksara** [pikepdf/pikepdf#567] [Q&A]
Hi , I would like to snip a area of a vector pdf to a new pdf. the external contents outside of the visible rectangle region shouldbe permanently deleted. how do i achieve this using pikepdf?
thanks for your advise.

**Comment by jbarlow83** [comment]
pikepdf does not have a vector graphics model that would be necessary to do this and is unlikely to ever get one. It's a low level PDF editing library but does not have native rendering facilities, except for some basic ones.

You could crop to size and use Ghostscript, or perhaps, PyMuPDF.
